internal/core/items/embed.rs

```diff
@@ -0,0 +1,145 @@
+// Copyright © SixtyFPS GmbH <info@slint-ui.com>
+// SPDX-License-Identifier: GPL-3.0-only OR LicenseRef-Slint-commercial
+
+/*!
+This module contains the builtin Embed related items
+
+When adding an item or a property, it needs to be kept in sync with different place.
+Lookup the [`crate::items`] module documentation.
+*/
+use super::{Item, ItemConsts, ItemRc, RenderingResult};
+use crate::component::{ComponentRc, ComponentWeak};
+use crate::input::{
+    FocusEvent, FocusEventResult, InputEventFilterResult, InputEventResult, KeyEvent,
+    KeyEventResult, MouseEvent,
+};
+use crate::item_rendering::CachedRenderingData;
+use crate::layout::{LayoutInfo, Orientation};
+use crate::lengths::{LogicalLength, LogicalPoint, LogicalRect, LogicalSize};
+use crate::properties::PropertyTracker;
+#[cfg(feature = "rtti")]
+use crate::rtti::*;
+use crate::window::WindowAdapter;
+use crate::Property;
+use alloc::rc::Rc;
+use const_field_offset::FieldOffsets;
+use core::cell::RefCell;
+use core::pin::Pin;
+use i_slint_core_macros::*;
+
+#[repr(C)]
+#[derive(FieldOffsets, Default, SlintElement)]
+#[pin]
+/// The implementation of the `Image` element
+pub struct Embed {
+    pub component: Property<Option<ComponentRc>>,
+
+    pub x: Property<LogicalLength>,
+    pub y: Property<LogicalLength>,
+    pub width: Property<LogicalLength>,
+    pub height: Property<LogicalLength>,
+
+    pub cached_rendering_data: CachedRenderingData,
+
+    component_tracker: RefCell<Option<Pin<Box<PropertyTracker>>>>,
+    component_rc: RefCell<Option<ComponentRc>>,
```

Comment:
Sadly `RefCell` isn't `repr(C)`, so it can't be used here. I think that you may have to do a heap allocation here like we do for the Flickable, along with accessor functions in Rust in `Embed` and free-standing C++ functions that forward.